README.md

```diff
@@ -643,340 +531,322 @@ terragrunt = {
+
+
+
+
+### Work with multiple Terraform modules
+
+
+#### Motivation
+
+Let's say your infrastructure is defined across multiple Terraform modules:
 
 ```
-my-terraform-repo
-  └ terraform.tfvars
-  └ stage
-      └ frontend-app
-          └ main.tf
-          └ terraform.tfvars
-      └ backend-app
-          └ main.tf
-          └ terraform.tfvars
-      └ search-app
-          └ main.tf
-          └ terraform.tfvars
-      └ mysql
-          └ main.tf
-          └ terraform.tfvars
-      └ redis
-          └ main.tf
-          └ terraform.tfvars
-      └ vpc
-          └ main.tf
-          └ terraform.tfvars
+root
+├── backend-app
+│   └── main.tf
+├── frontend-app
+│   └── main.tf
+├── mysql
+│   └── main.tf
+├── redis
+│   └── main.tf
+└── vpc
+    └── main.tf
 ```
 
-There is one module to deploy a frontend-app, another to deploy a backend-app, another for the MySQL database, and so 
-on. To deploy such an environment, you'd have to manually run `terragrunt apply` in each of the subfolders. How do you
-avoid this tedious and time-consuming process?
+There is one module to deploy a frontend-app, another to deploy a backend-app, another for the MySQL database, and so
+on. To deploy such an environment, you'd have to manually run `terraform apply` in each of the subfolder, wait for it 
+to complete, and then run `terraform apply` in the next subfolder. How do you avoid this tedious and time-consuming 
+process?
+
+
+#### The apply-all, destroy-all, and output-all commands
 
-The answer is that you can use the `apply-all` command:
+To be able to deploy multiple Terraform modules in a single command, add a `terraform.tfvars` file to each module:
+
+```
+root
+├── backend-app
+│   ├── main.tf
+│   └── terraform.tfvars
+├── frontend-app
+│   ├── main.tf
+│   └── terraform.tfvars
+├── mysql
+│   ├── main.tf
+│   └── terraform.tfvars
+├── redis
+│   ├── main.tf
+│   └── terraform.tfvars
+└── vpc
+    ├── main.tf
+    └── terraform.tfvars
+```
+
+Inside each `terraform.tfvars` file, add a `terragrunt = { ... }` block to identify this as a module managed by 
+Terragrunt (the block can be empty or include any of the configs described in this documentation):
+
+```hcl
+terragrunt = {
+  # Put your Terragrunt configuration here
+}
+```
+
+Now you can go into the root folder and deploy all the modules within it by using the `apply-all` command:
  
 ```
-cd my-terraform-repo/stage
+cd root
 terragrunt apply-all
 ```
 
-When you run this command, Terragrunt will find all `terraform.tfvars` files in the subfolders of the current working 
-directory that contain `terragrunt = { ... }` blocks, and run `terragrunt apply` in each one concurrently.
+When you run this command, Terragrunt will look through all the subfolders of the current working directory, find all 
+`terraform.tfvars` files with a `terragrunt = { ... }` block, and run `terragrunt apply` in each one concurrently.
```

Comment:
That includes recursive subfolders. I added the word "recursive" to clarify: https://github.com/gruntwork-io/terragrunt/pull/167/commits/5af50b0d64c2cf113a35458edbac78085597401c